google/cloud/bigquery/v2/minimal/internal/dataset_logging_test.cc

```diff
@@ -78,29 +64,25 @@ TEST_F(DatasetLoggingClientTest, GetDataset) {
                 std::move(http_response));
           });
 
-  // Not checking all fields exhaustively since this has already been tested.
-  // Just ensuring here that the request and response are being logged.
-  EXPECT_CALL(*log_backend_, ProcessWithOwnership)
-      .WillOnce([](LogRecord const& lr) {
-        EXPECT_THAT(lr.message, HasSubstr(" << "));
-        EXPECT_THAT(lr.message, HasSubstr(R"(GetDatasetRequest)"));
-        EXPECT_THAT(lr.message, HasSubstr(R"(project_id: "p-id")"));
-        EXPECT_THAT(lr.message, HasSubstr(R"(dataset_id: "d-id")"));
-      })
-      .WillOnce([](LogRecord const& lr) {
-        EXPECT_THAT(lr.message, HasSubstr(R"(GetDatasetResponse)"));
-        EXPECT_THAT(lr.message, HasSubstr(R"(project_id: "p-id")"));
-        EXPECT_THAT(lr.message, HasSubstr(R"(dataset_id: "d-id")"));
-      });
-
   auto client = CreateMockDatasetLogging(std::move(mock_stub));
   GetDatasetRequest request("p-id", "d-id");
   rest_internal::RestContext context;
 
   client->GetDataset(context, request);
+
+  auto actual_lines = log.ExtractLines();
+
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(GetDatasetRequest)")));
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(project_id: "p-id")")));
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(dataset_id: "d-id")")));
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(GetDatasetResponse)")));
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(project_id: "p-id")")));
+  EXPECT_THAT(actual_lines, Contains(HasSubstr(R"(dataset_id: "d-id")")));
```

Comment:
The expectation here is that the log response contains the relevant information that is considered important in this case "project-id" and "dataset-id" so from that perspective it is testing that.

To me ordering of the substrings are not as important as the fact that the log contains that information.

Reply:
I'm afraid I can't see how that is related to my concern, so I guess I did a poor job expressing it.  Sorry.

I also don't think ordering is that important, but it could be one way to distinguish between (a) expecting two matches, which is meaningful, and (b) expecting a match twice, which is just redundant.

Reply:
Sorry I am confused as to what change you want here if ordering is not important as you mention.

Please state explicitly what is needed to change here. If you're referring to a specific matcher then please state which matcher to use. Thanks

Reply:
OK, the explicit suggestion here is:  Remove lines 79 and 80, as they just repeat lines 76 and 77 respectively.

Then, also remove similar redundancies from a few other places.